src/som/compiler/Parser.java

```diff
@@ -1444,6 +1450,31 @@ private LiteralNode literalString() throws ParseError {
     return new StringLiteralNode(s, getSource(coord));
   }
 
+  private LiteralNode literalArray(final MethodBuilder builder) throws ProgramDefinitionError {
+    SourceCoordinate coord = getCoordinate();
+    List<ExpressionNode> expressions = new ArrayList<ExpressionNode>();
+
+    expect(NewLiteralArray, DelimiterOpeningTag.class);
+
+    boolean sawPeriod = true;
+
+    while (true) {
+      comments();
+
+      if (sym == EndLiteralArray) {
+        expect(EndLiteralArray, DelimiterClosingTag.class);
+        return createLiteralArray(expressions, getSource(coord));
+      }
+
+      if (!sawPeriod) {
+        expect(Period, null);
+      }
+
+      expressions.add(expression(builder));
+      sawPeriod = accept(Period, StatementSeparatorTag.class);
+    }
```

Comment:
Hmm good idea, it would be nice to make the method more concise. However, unless I've misunderstood, I believe that this suggestion would throw an error on an empty literal array: `{}`. One option would be to check for `EndLiteralArray` before the loop, which should work.

Is there anything that can go into the literal array that won't parse as an expression? If not, then I will go ahead and make the change!